specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/bastionHost.json

```diff
@@ -0,0 +1,440 @@
+                "BastionHosts"
+             ],
+             "operationId":"BastionHosts_ListBySubscription",
+             "description":"Lists all Bastion Hosts in a subscription.",
+             "parameters":[  
+                {  
+                   "$ref":"./network.json#/parameters/ApiVersionParameter"
+                },
+                {  
+                   "$ref":"./network.json#/parameters/SubscriptionIdParameter"
+                }
+             ],
+             "responses":{  
+                "200":{  
+                   "description":"Success. The operation returns a list of BastionHost resources.",
+                   "schema":{  
+                      "$ref":"#/definitions/BastionHostListResult"
+                   }
+                }
+             },
+             "x-ms-examples":{  
+                "List all Bastion Hosts for a given subscription":{  
+                   "$ref":"./examples/BastionHostListBySubscription.json"
+                }
+             },
+             "x-ms-pageable":{  
+                "nextLinkName":"nextLink"
+             }
+          }
+       },
+       "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/bastionHosts":{  
+          "get":{  
+             "tags":[  
+                "BastionHosts"
+             ],
+             "operationId":"BastionHosts_ListByResourceGroup",
+             "description":"Lists all Bastion Hosts in a resource group.",
+             "parameters":[  
+                {  
+                   "name":"resourceGroupName",
+                   "in":"path",
+                   "required":true,
+                   "type":"string",
+                   "description":"The name of the resource group."
+                },
+                {  
+                   "$ref":"./network.json#/parameters/ApiVersionParameter"
+                },
+                {  
+                   "$ref":"./network.json#/parameters/SubscriptionIdParameter"
+                }
+             ],
+             "responses":{  
+                "200":{  
+                   "description":"Success. The operation returns a list of BastionHost resources.",
+                   "schema":{  
+                      "$ref":"#/definitions/BastionHostListResult"
+                   }
+                }
+             },
+             "x-ms-examples":{  
+                "List all Bastion Hosts for a given resource group":{  
+                   "$ref":"./examples/BastionHostListByResourceGroup.json"
+                }
+             },
+             "x-ms-pageable":{  
+                "nextLinkName":"nextLink"
+             }
+          }
+       },
+       "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/{virtualNetworkName}/bastionHosts/default":{  
+          "get":{  
+             "tags":[  
+                "BastionHosts"
+             ],
+             "operationId":"BastionHosts_ListVnet",
```

Comment:
Possible it make sense to change this to `ListByVirtualNetwork`. The following considerations there:
- ListVnet seems to suppose list of virtual networks
- Other list operations e.g. ListByResourseGroup contain full name

+@sergey-shandar to discuss further

Reply:
Fine with ListByVirtualNetwork. We are going to deprecate this API in future as there are RBAC issues. Should i just remove it for now?

Reply:
I believe it is better to discuss with your management or someone who better knows this feature and further plans on using/updating it.
In general deprecating of the released feature could be more complex and time-taking than just not-adding it.
+@dsgouda, @sergey-shandar

Reply:
Removing this API as we are confident that we are going to move away from this